### ISSUE TITLE
[Feature Request] Extra field "title" for personal client information

### ISSUE THREAD

**Issue by onny** [InvoicePlane/InvoicePlane#1003]
It would be nice to have an extra field in the personal information area of a client for specifying a title.
This should get rendered in the invoice pdf in the billing to area:

Mr.
Christian Bauer
Example Street 1
Example City
etc.

The extra field could be a drop down for common titles like "Mr" or "Mrs" but also should provide a free form text field in case one would need to specify a complex title (including academic title etc).

**Comment by nielsdrost7** [comment]
So a select box with a list with contents:
[array of settings like Mr., Mrs, etc] AND a list-item: Custom (or 'type it yourself).

Save the chosen title to "create field in `clients` table

**Comment by nielsdrost7** [comment]
Yes we can, but it would still be a select box with that list of possibilities.
Plus I know someone who wants to manipulate it through the ipconfig file.

@KevinJ30 would you like to make a branch with the suggested solution? For example with your suggestion of that enum you were talking about

**Comment by nielsdrost7** [comment]
@onny I've improved the Pull-Request that @KevinJ30 made. Could you take a look?
https://github.com/InvoicePlane/InvoicePlane/pull/1101

**Comment by nielsdrost7** [comment]
@onny Merged, will be released in version 1.6.2
can you check please in the Beta version (coming up soon) if this is what you meant?

**Comment by onny** [comment]
Thank you @nielsdrost7, going to test it today. Just reworked the InvoicePlane package in NixOS so it will be easier to test development branches https://github.com/NixOS/nixpkgs/pull/353783 :)

**Comment by nielsdrost7** [comment]
> Just reworked the InvoicePlane package in NixOS so it will be easier to test development branches

That's awesome, man.
By the way: I just thought of something.
Would you like the values in the ClientTitle to be configurable?
Maybe a setting in ipconfig.php?
Comma separated, lowercase?

If my colleagues won

**Comment by onny** [comment]
Just tried the new feature, seems to have some issues: After clicking on "Add client" I get: 

``` 

An uncaught Exception was encountered

Type: TypeError

Message: Clients::{closure}(): Argument #1 ($clientTitleEnum) must be of type ClientTitleEnum, stdClass given

Filename: /nix/store/ngx36v3bsk8cxwdngcxbcc0rb7v34jbl-invoiceplane-localhost-1.6.1/application/modules/clients/controllers/Clients.php

Line Number: 272

Backtrace:

File: /nix/store/ngx36v3bsk8cxwdngcxbcc0rb7v34jbl-invoiceplane-localhost-1.6.1/application/modules/clients/controllers/Clients.php
Line: 273
Function: array_map

File: /nix/store/ngx36v3bsk8cxwdngcxbcc0rb7v34jbl-invoiceplane-localhost-1.6.1/application/modules/clients/controllers/Clients.php
Line: 184
Function: get_client_title_choices

File: /nix/store/ngx36v3bsk8cxwdngcxbcc0rb7v34jbl-invoiceplane-localhost-1.6.1/index.php
Line: 329
Function: require_once
```

**Comment by nielsdrost7** [comment]
Ok, I'll take a look. I'll let you knkw

**Comment by onny** [comment]
Looks like the dropdown always falls back to "Mr" and doesn't store the change. Also this gets always rendered as "Custom [Erika Musterfrau]".

If I select the "Custom" field I'm unable to enter something custom like "Frau Prof. Dr."

![20241110_15h49m28s_grim](https://github.com/user-attachments/assets/4637d669-0263-457e-ae73-946074ed5d90)
![20241110_15h48m54s_grim](https://github.com/user-attachments/assets/d3afb897-c74c-46a4-8a89-f69371c797b4)
![20241110_15h48m26s_grim](https://github.com/user-attachments/assets/b854b4ba-9a87-42c3-9761-801dc608f1db)

**Comment by nielsdrost7** [comment]
> Looks like the dropdown always falls back to "Mr" and doesn't store the change. Also this gets always rendered as "Custom [Erika Musterfrau]".

My mistake again. Something went wrong while merging.
The real title field is at the bottom of the screen, near the gender.

I want to make a case for `Custom` by default:
Let's say my client is "Localhost Corporation"
Then you don't want to give it a title.

Right now (after I've changed the ClientTitle enum)
When you keep it on `Custom`  and "empty"
The title won't be rendered. That's what you want with companies as clients, correct?

**Comment by onny** [comment]
works now and is really nice to use :+1: 

![image](https://github.com/user-attachments/assets/0fe58b61-a6ab-4bb5-a5d2-416687558776)

**Comment by nielsdrost7** [comment]
Good to know man! We'll release a beta-version soon

**Comment by sudwebdesign** [comment]
Hi, See #1150 to add translation possibility for Client Title options

**Comment by nielsdrost7** [comment]
Seen it and merged it.

**Comment by nielsdrost7** [comment]
Solved by [releasing version 1.6.2](https://github.com/InvoicePlane/InvoicePlane/releases/tag/v1.6.2)

**Comment by onny** [comment]
Thank you, really great! Going to package the update for NixOS now :)

**Comment by getafixza** [comment]
Oh no, I did not.  

xam-ps, sincerely thank you for your reply and advice.

Okey I now revert my changes, then edit the ip_config.php file by changing DISABLE_SETUP=true to false so that it is in setup mode, and the link /index.php/setup/upgrade_tables will run. Thanks a million it worked.
Excuse I did not seem to see the instruction to run /index.php/setup/upgrade_tables after the update. It's a pitty that it is not updating automatically. E.g. when the app see there is a version difference between the new files and the database, that it would automatically do the update. I hope my suggestion make sense.

Kind Regards
